sklearn/tests/test_calibration.py

```diff
@@ -341,3 +349,53 @@ def decision_function(self, X):
     calibrated_clf = CalibratedClassifierCV(MockTensorClassifier())
     # we should be able to fit this classifier with no error
     calibrated_clf.fit(X, y)
+
+
+# Create prefit pipeline
+word_data = [
+    {'state': 'NY', 'age': 'adult'},
+    {'state': 'TX', 'age': 'adult'},
+    {'state': 'VT', 'age': 'child'},
+]
+word_labels = [1, 0, 1]
+
+pipeline_prefit = Pipeline([
+    ('vectorizer', DictVectorizer()),
+    ('clf', RandomForestClassifier())
+])
+pipeline_prefit.fit(word_data, word_labels)
+
+
+def test_calibration_pipeline():
+    # Test that calibration works in prefit pipeline with transformer,
+    # where `X` is not array-like, sparse matrix or dataframe at the start.
+    # See issue #8710
+    calib_clf = CalibratedClassifierCV(pipeline_prefit, cv='prefit')
+    calib_clf.fit(word_data, word_labels)
+    # Check attributes are obtained from fitted estimator
+    assert_array_equal(calib_clf.classes_, [0, 1])
```

Comment:
```suggestion
    assert_array_equal(calib_clf.classes_, [0, 1])
    assert not hasattr(calib_clf, "n_features_in_")
```

Reply:
I test this later in:
```
    calib_clf = CalibratedClassifierCV(pipeline_prefit, cv='prefit')
    msg = "'CalibratedClassifierCV' object has no attribute 'n_features_in_'"
    with pytest.raises(AttributeError, match=msg):
        calib_clf.n_features_in_
```
but I can amend to your way as it is shorter?